src/Lexer.js

```diff
@@ -350,7 +350,8 @@ export class Lexer {
 
     // Mask out escaped em & strong delimiters
     while ((match = this.tokenizer.rules.inline.escapedEmSt.exec(maskedSrc)) != null) {
-      maskedSrc = maskedSrc.slice(0, match.index) + '++' + maskedSrc.slice(this.tokenizer.rules.inline.escapedEmSt.lastIndex);
+      maskedSrc = maskedSrc.slice(0, match.index + match[0].length - 2) + '++' + maskedSrc.slice(this.tokenizer.rules.inline.escapedEmSt.lastIndex);
+      this.tokenizer.rules.inline.escapedEmSt.lastIndex--;
```

Comment:
@UziTech Similar to my [comment](https://github.com/markedjs/marked/pull/2628/files#r1008978064) on #2628, would this whole PR be cleaner if we just pre-masked out *all* escaped characters (not just `*` and `_`)? Rather than each tokenizer trying to handle escaped characters in its own way?

Reply:
I think that would require a lot more work trying to make sure we mask out the right characters everywhere. #2628 and this PR are two separate PRs doing two different things. I think we should merge this PR pretty much how it is. If we want to consolidate how things are done after merging this we can.